### PR TITLE
added/updated tests

### PR DESCRIPTION
**Pull request by jdhudson3**

### CODE REVIEW

File: src/test/java/com/bandwidth/sdk/messaging/MessagingClientTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            .time("timestamp")
+            .build();
+
+
+    @Test
+    public void testMessagingClient() throws ExecutionException, IOException, InterruptedException {
+        when(mockClient.preparePost(ArgumentMatchers.anyString())).thenReturn(boundRequestBuilder);
+        when(boundRequestBuilder.setBody(ArgumentMatchers.anyString())).thenReturn(boundRequestBuilder);
+        when(boundRequestBuilder.execute()).thenReturn(listenableFuture);
+        when(listenableFuture.toCompletableFuture()).thenReturn(CompletableFuture.completedFuture(response));
```
Comment (jmulford-bandwidth) [inline src/test/java/com/bandwidth/sdk/messaging/MessagingClientTest.java]: I'm assuming this covers the final .get() call in this code right?
```
        Response response = httpClient.preparePost(url)
                .setBody(messageSerde.serialize(request))
                .execute().toCompletableFuture().get();
```

Comment (jdhudson3) [reply src/test/java/com/bandwidth/sdk/messaging/MessagingClientTest.java]: Line 69 covers that. 65-68 are so it doesn't throw a NullPointerException before getting there.

Comment (jmulford-bandwidth) [reply src/test/java/com/bandwidth/sdk/messaging/MessagingClientTest.java]: Cool

File: src/main/java/com/bandwidth/sdk/messaging/MessagingClient.java [hunk 2]
```diff
@@ -43,6 +43,11 @@ public MessagingClient(String userId, String token, String secret) {
         httpClient = asyncHttpClient(httpClientConfig);
     }
 
+    MessagingClient(String userId, AsyncHttpClient mockClient) {
```
Comment (fuchsnj) [inline src/main/java/com/bandwidth/sdk/messaging/MessagingClient.java]: add @VisibleForTesting so people understand why it's `package-private`

Comment (fuchsnj) [reply src/main/java/com/bandwidth/sdk/messaging/MessagingClient.java]: You could also just make this a setter

File: src/test/java/com/bandwidth/sdk/messaging/MessagingCallbackHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            .build();
+
+    private final MessagingCallbackHelper callbackHelper = new MessagingCallbackHelper();
+    private final MessageSerde serde = new MessageSerde();
+
+
+    @Test
+    public void testParseCallback() throws IOException {
+        MessageEvent returnedEvent = callbackHelper.parseCallback(serde.serialize(messageEvent));
+        Assert.assertEquals(returnedEvent,messageEvent);
```
Comment (fuchsnj) [inline src/test/java/com/bandwidth/sdk/messaging/MessagingCallbackHelperTest.java]: What do you think about using AssertJ for our assertions? This is what we use is most other projects, and they tend to read better / provide a bit more features
http://joel-costigliola.github.io/assertj/

**Pull request merged**